### PR TITLE
Convert a put to to a delete operation in ReadWriteTable and TableWriteFunction when input value is null

### PR DESCRIPTION
**Pull request by weisong44**
Currently, the behavior of putting a null value is inconsistent: it is a delete for RocksDB, and not supported in in-memory store, and on a case-by-case basis for remote tables. It is desirable to unify the behavior. Furthermore, it eases the writing of a change captured stream to a table. A change captured stream contains typically 3 types of events: INSERT, UPDATE and DELETE, and they need to be applied properly when written to a table to produce a correct snapshot. In a change captured stream the payload of a DELETE event is typically is null, and this would result in a delete operation to a table in sendTo() operator.

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/table/caching/guava/GuavaCacheTable.java [hunk 1]
```diff
@@ -19,19 +19,17 @@
 
 package org.apache.samza.table.caching.guava;
 
+import com.google.common.cache.Cache;
```
Comment (pdu-mn1) [inline samza-core/src/main/java/org/apache/samza/table/caching/guava/GuavaCacheTable.java]: nit: unnecessary change

File: samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java [hunk 2]
```diff
@@ -19,8 +19,8 @@
 
 package org.apache.samza.table.remote;
 
+import com.google.common.base.Preconditions;
```
Comment (pdu-mn1) [inline samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java]: nit: unnecessary change

File: samza-core/src/main/java/org/apache/samza/table/remote/TableWriteFunction.java [hunk 3]
```diff
@@ -44,13 +43,19 @@
 public interface TableWriteFunction<K, V> extends Serializable, InitableFunction, ClosableFunction {
   /**
    * Store single table {@code record} with specified {@code key}. This method must be thread-safe.
+   *
+   * The key is deleted, if record is {@code null}.
```
Comment (pdu-mn1) [inline samza-core/src/main/java/org/apache/samza/table/remote/TableWriteFunction.java]: nit: add "its corresponding" to match the previous doc

File: samza-kv/src/main/java/org/apache/samza/storage/kv/LocalStoreBackedReadWriteTable.java [hunk 4]
```diff
@@ -19,7 +19,6 @@
 package org.apache.samza.storage.kv;
 
 import java.util.List;
-
```
Comment (pdu-mn1) [inline samza-kv/src/main/java/org/apache/samza/storage/kv/LocalStoreBackedReadWriteTable.java]: nit: unnecessary change

File: samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java [hunk 5]
```diff
@@ -30,9 +30,7 @@
 import org.apache.samza.task.TaskContext;
 import org.apache.samza.util.RateLimiter;
 
-import com.google.common.base.Preconditions;
-
-import static org.apache.samza.table.remote.RemoteTableDescriptor.RL_WRITE_TAG;
+import static org.apache.samza.table.remote.RemoteTableDescriptor.*;
```
Comment (pdu-mn1) [inline samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java]: nit: unnecessary change

File: samza-core/src/main/java/org/apache/samza/table/remote/TableWriteFunction.java (hunk 3 above)
Comment (weisong44) [reply samza-core/src/main/java/org/apache/samza/table/remote/TableWriteFunction.java]: There is only one key/value pair, hence no mentioning.

File: samza-api/src/main/java/org/apache/samza/table/ReadWriteTable.java [hunk 6]
```diff
@@ -36,17 +35,21 @@
   /**
    * Updates the mapping of the specified key-value pair; Associates the specified {@code key} with the specified {@code value}.
    *
+   * The key is deleted from the table, if value is {@code null}.
```
Comment (prateekm) [inline samza-api/src/main/java/org/apache/samza/table/ReadWriteTable.java]: Nitpick: no comma before if everywhere.

Comment (weisong44) [reply samza-api/src/main/java/org/apache/samza/table/ReadWriteTable.java]: +1

**Comment by weisong44** [comment]
I've updated the PR based on comments, can you kindly merge it?
